### ISSUE TITLE
pypi version

### ISSUE THREAD

**Issue by Pantelispanka** [openmm/pdbfixer#254]
Will there ever be a pypi package for the module ? If you are planning for that i can offer some help! Thanks!

**Comment by peastman** [comment]
I hope so!  We first need PyPI packages for OpenMM, which PDBFixer depends on.  We're investigating that right now: https://github.com/openmm/openmm/issues/3796.  If you want to help, that would be fantastic.

**Comment by minhuanli** [comment]
Hi, I want to do a quick follow up on this issue. 

Why don't we make `PDBFixer` a PyPI package alone? The main OpenMM module seems tricky because of the cuda stuff, but you could assume people have already installed the OpenMM before they need PDBFixer? Just save people from downloading the whole repo before installation.

**Comment by peastman** [comment]
What would be the use case for that?  If you've already installed OpenMM with conda, what's the benefit of installing PDBFixer with pip instead of conda?

**Comment by minhuanli** [comment]
One common case for me is that, I have installed OpenMM according to the official document:

```
conda install -c conda-forge openmm cudatoolkit=xx.x
```

Then I realize that I also need PDBfixer to preprocess PDB files. 

It would be great if I could simply do `pip install PDBFixer` instead of git clone the whole repo and run `pip install .`

**Comment by peastman** [comment]
Correct.  This will install it, including all dependencies.

```
conda install -c conda-forge pdbfixer
```

We should add that to the README.

**Comment by minhuanli** [comment]
Got it! That makes much more sense now. Yes, should add this to the README. Thank you for the clarification.

**Comment by peastman** [comment]
The updated instructions are in #265.

**Comment by recursingfeynman** [comment]
If pdbfixer were on pypi it would be convenient to install it along with dependencies via poetry. Poetry can also handle dependencies like conda, but unlike it it shows better performance

**Comment by k9cdt** [comment]
Now that OpenMM has been available on PyPI since May 10, 2024, I think it’s a good time to bring this issue up again.

Uploading pdbfixer to PyPI would really help those of us using it as a dependency in our own projects. Currently, any package on PyPI can’t depend on something that isn’t also on PyPI (which I get, even though it’s a bit limiting).

[PEP 725](https://peps.python.org/pep-0725/) is still a draft, so it looks like we’ll have to wait before GitHub repositories can be used as dependencies.

**Comment by peastman** [comment]
The current PyPI builds for OpenMM are just a proof of concept and shouldn't be relied on.  I'm hoping in the future we'll have proper builds that are fully tested and supported.  We have a ways to go before that can happen though.

**Comment by kiyoon** [comment]
@peastman I've been using the pypi builds of OpenMM and it's been working solidly. When can I expect to see Pdbfixer as well? I guess this repository is pure python and does not require build steps, so it should be easy to upload it as a proof of concept too, right?

**Comment by peastman** [comment]
Once OpenMM 8.2 is released and we've confirmed the pip packages work correctly, PDBFixer will be one of the next ones to do.

**Comment by yaideltg** [comment]
@peastman I would like to add to the discussion that, in my opinion, having a PyPI package has many advantages. Even when it depends on others that need to be installed manually. For example, if I'm using PDBfixer as a dependency to a package I'm developing, it makes a big difference to have it accessible via PyPI. This way, my users would only have to worry about installing OpenMM. And although some would say that if you already installed OpenMM manually, then you will have no problems installing PDBfixer, I think that's not correct: following this reasoning, package indexes and managers existence would make no sense, and it does, a lot.

Also, I think we should not force packages to be only accessible via conda. Why would I need to install conda if I just want to use PDBfixer?

Finally, OpenMM 8.2.0 it's been out since Nov 8, 2024. Is there any plan to finally make PDBfixer a PyPI package? I could eventually help too.

**Comment by kiyoon** [comment]
Apparently the name is reserved for some reason and can't be used on pypi.

However, you can just take another name. I would go for "openmm-pdbfixer" and the package name can stay the same as "pdbfixer".  Once you resolve the issue with the reserved name you can also migrate to there then.

**Comment by peastman** [comment]
See https://github.com/pypi/support/issues/5093.  We've been blocked for months waiting on the PyPA team.

The problem with posting it under a different name is that people start depending on that name.  Then we're stuck supporting it forever.

**Comment by kiyoon** [comment]
You're right. Personally I don't have a problem waiting. Hope it goes all well.

**Comment by milot-mirdita** [comment]
I also asked a second degree contact to forward this issue to someone at PyPi. They forwarded it but also the general advice was:
> ... publishing under a different name may be the best path forward. (You can even keep the `import pdbfixer`, in Python project name and module name can be different)
